### PR TITLE
Update readme for clarity and added git info

### PR DESCRIPTION
**Pull request by tyler-eon**
We want the instructions to be as clear as possible so that candidates don't unnecessarily get hung up on certain details and focus on the right aspects of the challenge.

Additionally, we want candidates to show their git skills if they have any, so we're adding instructions to create a git repo for the submission if the candidate knows and can use git to do so.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,75 @@
...
+1. Your work assets, as one or both of the following:
+  - A link to a hosted git repository.
+  - A ZIP file containing your project directory. Include the `.git` sub-directory if you used git.
+2. Proof of completion, as one or both of the following:
+  - A link to a hosted AWS deployment.
+  - One or more screenshots showing, at least, the index page of the final deployment.
+
+Your work assets should include:
+
+- Terraform or CloudFormation modules, if you completed that task.
```
Comment (milin82) [inline README.md]: I suggest we add AWS CDK to the list here as well.

Comment (tyler-eon) [reply README.md]: Added CDK.

Side note: we might want to eventually think about letting people host this on at least Google Cloud as well, if they choose to do so, as it's still relatively similar to AWS. I wouldn't mind allowing Azure deployments either but it does differ significantly from AWS and GCP.

**Review by milin82 — APPROVED** [review]
looks good to me

**Comment by rahulsen** [comment]
Since we added CDK as an option, there are a few more places where we should mention it (where tf and cf are already mentioned) for completeness. Or just shorten them all to "IaC" in all but one place.

**Comment by rahulsen** [comment]
Testing this on gcp is definitely a good idea. We will have to update detection mechanisms to account for gcp.

**Comment by tyler-eon** [comment]
> Since we added CDK as an option, there are a few more places where we should mention it (where tf and cf are already mentioned) for completeness. Or just shorten them all to "IaC" in all but one place.

Done. I like the idea of using just "IaC" and I've also expanded the definition to include _whichever_ IaC of choice the candidate is comfortable with. @rahulsen would you agree that if someone can learn and use, e.g., CloudFormation successfully then they could learn and use Terraform successfully? That's the assumption I am thinking we can make by letting them choose whatever IaC path they want. However, I am specifically calling out Terraform as being preferred since most companies want to move in that direction and it's becoming a cloud-agnostic choice for many. Thoughts?

**Review by rahulsen — APPROVED** [review]

**Pull request merged**